tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs

```diff
@@ -887,6 +889,63 @@ private async Task UploadTestRunBlobAsync(string account, Build build, TestRun t
             }
         }
 
+        private async Task UploadTestRunResultsAsync(string account, Build build, TestRun testRun)
+        {
+            try
+            {
+                var continuationToken = string.Empty;
+                var buildIds = new[] { build.Id };
+                var discoveredTestResults = new List<ShallowTestCaseResult>();
+                var blobPath = $"{build.Project.Name}/{testRun.CompletedDate:yyyy/MM/dd}/{testRun.Id}/results.jsonl";
+                var blobClient = this.testRunsContainerClient.GetBlobClient(blobPath);
+
+                if (await blobClient.ExistsAsync())
+                {
+                    this.logger.LogInformation("Skipping existing test results blob for build {BuildId}, test run {RunId}", build.Id, testRun.Id);
+                    return;
+                }
+
+                do
+                {
+                    // the unfortunate fact of the matter is that testResultsClient does not have a API Surface that can get us
```

Comment:
I was able to find a method on the TestResultsHttpClientBase class that matches the api you want:
```c#
public virtual Task<List<TestCaseResult>> GetTestResultsAsync(string project, int runId, ResultDetails? detailsToInclude = null, int? skip = null, int? top = null, IEnumerable<TestOutcome> outcomes = null, bool? newTestsOnly = null, object userState = null, CancellationToken cancellationToken = default(CancellationToken))
```

The return type has more detail than from `GetTestResultsByBuildWithContinuationTokenAsync`.  We need some date field in the table and TestCaseResult contains StartedDate, CompletedDate, and FailingSince (object that includes a build reference and date)

Reply:
Working through the schema changes now. Thanks for the suggestion!

Reply:
Alrighty, this is what the schema looks like now:

```
.create-merge table TestRunResult (
    OrganizationName: string,
    ProjectId: string,
    ProjectName: string,
    BuildDefinitionId: int,
    BuildDefinitionPath: string,
    BuildDefinitionName: string,
    BuildId: int,
    TestRunId: int,
    TestCaseId: int, 
    TestCaseReferenceId: int,
    TestCaseTitle: string,
    Outcome: string,
    Priority: int,
    AutomatedTestName: string,
    AutomatedTestStorageName: string,
    FailingSince: dynamic,
    FailureType: string,
    StartedDate: datetime,
    CompletedDate: datetime,
    EtlIngestDate: datetime
)
```